### PR TITLE
Improve data splitting implementation

### PR DESCRIPTION
**Pull request by gchindemi**
This pull request simplifies the dataset handling logic by removing support for train/dev/test splits and consolidating record management. It also refactors related functions and improves the clarity of the codebase. The most important changes are grouped below by theme.

### Dataset Handling Simplification:
* Removed support for `dev_ratio`, `test_ratio`, `dev_seed`, and `test_seed` parameters in `load_records`, along with associated logic for splitting datasets into train/dev/test groups. The function now returns a single list of records.
* Updated the return type of `load_records` to reflect the simplified structure.
* Removed the dependency on `train_test_split` from `src/lisbet/datasets/core.py`.

### Refactoring and Consolidation:
* Renamed `group_records` to `records` in `evaluate_model` and `_process_inference_dataset` to align with the simplified dataset structure.
* Refactored `_load_records` in `src/lisbet/training.py` to `_load_multi_records` and introduced `_split_multi_records` for handling train/dev splits at the task level.

### Parameter and Argument Cleanup:
* Removed the `--seed_test_split` argument from the CLI parser and updated the help text for `--train_sample` and `--dev_ratio` arguments for clarity.
* Removed unused parameters (`test_ratio`, `dev_ratio`, `test_seed`, `dev_seed`) from `_generate_seeds` and `_load_records` in `src/lisbet/training.py`.

### Code Simplification:
* Replaced the `all_records` variable with `records` in `load_records` and removed redundant filtering logic. 
* Removed test set handling from `_configure_classification_task` and `_configure_selfsupervised_task`. 

### Dependency Updates:
* Added `train_test_split` back to `src/lisbet/training.py` for use in `_split_multi_records`.

These changes collectively simplify the dataset handling logic, reduce code complexity, and align the codebase with the updated requirements.

### CODE REVIEW

File: src/lisbet/training.py [hunk 1]
```diff
@@ -158,39 +130,72 @@ def _load_records(
...
+
+
+def _split_multi_records(
+    multi_records,
+    dev_ratio,
+    dev_seed,
+    task_ids,
+    task_data,
+):
+    """Split records into train and dev sets."""
```
[the remark this answers is not in the document]
Comment (gchindemi) [reply src/lisbet/training.py:143]: Not worth it now.

File: tests/test_training_helpers.py [hunk 2]
```diff
@@ -7,8 +7,77 @@
 import torch
 import xarray as xr
 import yaml
+from sklearn.utils._param_validation import InvalidParameterError
 
 import lisbet.training as training
+from lisbet.training import _split_multi_records
+
+
+@pytest.fixture
```
[the remark this answers is not in the document]
Comment (gchindemi) [reply tests/test_training_helpers.py:16]: Not worth it now.
